near-sdk/src/environment/env.rs

```diff
@@ -114,10 +115,24 @@ pub fn setup_panic_hook() {
 
 /// Reads the content of the `register_id`. If register is not used returns `None`.
 pub fn read_register(register_id: u64) -> Option<Vec<u8>> {
-    let len = register_len(register_id)?;
-    let res = vec![0u8; len as usize];
-    unsafe { sys::read_register(register_id, res.as_ptr() as _) };
-    Some(res)
+    // Get register length and convert to a usize. The max register size in config is much less
+    // than the u32 max so the abort should never be hit, but is there for safety because there
+    // would be undefined behaviour if the case would be hit.
+    let len: usize = register_len(register_id)?.try_into().unwrap_or_else(|_| abort());
```

Comment:
Is it really undefined behavior though? From the docs, casting just truncates:
> Casting from a larger integer to a smaller integer (e.g. u32 -> u8) will truncate

but it's not ideal either for us to have a truncated `len` either, so the change still seems valid to me

Reply:
The UB would happen not during truncation, but when the runtime writes past the truncated lenght.